x/poolmanager/routes.go

```diff
@@ -0,0 +1,569 @@
+package poolmanager
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	cosmwasmpooltypes "github.com/osmosis-labs/osmosis/v20/x/cosmwasmpool/types"
+	gammtypes "github.com/osmosis-labs/osmosis/v20/x/gamm/types"
+	"github.com/osmosis-labs/osmosis/v20/x/poolmanager/types"
+)
+
+var (
+	OSMO                 = "uosmo"
+	superfluidMultiplier = sdk.MustNewDecFromStr("1.5")
+	directRouteCache     map[string]uint64
+	spotPriceCache       map[string]osmomath.BigDec
+)
+
+func init() {
+	directRouteCache = make(map[string]uint64)
+	spotPriceCache = make(map[string]osmomath.BigDec)
+}
+
+// findDirectRoute finds all direct routes between two tokens
+func findDirectRoute(g types.RoutingGraphMap, start, end string) []*types.Route {
+	if innerMap, exists := g.Graph[start]; exists {
+		if routes, exists := innerMap.InnerMap[end]; exists {
+			for i := range routes.Routes {
+				routes.Routes[i].Token = end
```

Comment:
So we return the end token in every instance here since this is the "out" token. Imagine we are swapping across three pools, and each of these pools has more than two tokens. If we just stored the pool route (2>5>8), its much harder for someone to extrapolate what each out token needs to be. Since we can just mark this at time of route determination, this allows us to give the exact route (pool ID and token out) for each hop.

TL;DR it just makes it simpler when estimating the swap via the route if we note the out token of each hop in the event it has more than two tokens.

Reply:
I think that was you question, lmk if that answers it